src/modules/ft/data_provider/balance.rs

```diff
@@ -2,46 +2,52 @@ use crate::modules::ft;
 use crate::{db_helpers, rpc_helpers, types};
 use std::str::FromStr;
 
-// TODO PHASE 2 pagination (recently updated go first), by artificial index added to assets__fungible_token_events
 pub(crate) async fn get_ft_balances(
-    pool: &sqlx::Pool<sqlx::Postgres>,
+    pool_balances: &sqlx::Pool<sqlx::Postgres>,
     rpc_client: &near_jsonrpc_client::JsonRpcClient,
-    block: &db_helpers::Block,
     account_id: &near_primitives::types::AccountId,
-    pagination: &types::query_params::Pagination,
+    block: &db_helpers::Block,
+    // TODO pagination
+    limit: u32,
 ) -> crate::Result<Vec<ft::schemas::FtBalance>> {
+    // todo it's better to query by chunks, this list can bee potentially too big (500+ contracts)
     let query = r"
-        SELECT DISTINCT emitted_by_contract_account_id account_id
-        FROM assets__fungible_token_events
-        WHERE (token_old_owner_account_id = $1 OR token_new_owner_account_id = $1)
-            AND emitted_at_block_timestamp <= $2::numeric(20, 0)
-        ORDER BY emitted_by_contract_account_id
-        LIMIT $3::numeric(20, 0)
+        SELECT DISTINCT contract_account_id account_id
+        FROM coin_events
+        WHERE affected_account_id = $1
+        ORDER BY contract_account_id
     ";
     let contracts = db_helpers::select_retry_or_panic::<db_helpers::AccountId>(
-        pool,
+        pool_balances,
         query,
-        &[
-            account_id.to_string(),
-            block.timestamp.to_string(),
-            pagination.limit.to_string(),
-        ],
+        &[account_id.to_string()],
     )
     .await?;
+    tracing::info!(
```

Comment:
are you sure it is `info` level and not a `debug` one?

Reply:
I'd prefer to use `info` here because I want to see this message in prod
Prod usually doesn't track `debug` messages

I need this because I have issue related to this, and I want to understand how urgent it is

Reply:
I truly believe it should be `debug` after all. Do you mind adding the comment that it is temporary and referencing the issue? In case you're going to switch it after you make a decision about the urgency and the issue itself.

Reply:
After the issue is fixed, I'll drop it fully :)